pandas/core/arrays/string_arrow.py

```diff
@@ -763,6 +764,26 @@ def _str_contains(self, pat, case=True, flags=0, na=np.nan, regex=True):
         else:
             return super()._str_contains(pat, case, flags, na, regex)
 
+    def _str_startswith(self, pat, na=None):
+        if hasattr(pc, "match_substring_regex"):
+            result = pc.match_substring_regex(self._data, "^" + re.escape(pat))
```

Comment:
I don't think this will always be correct. Because of using this function, `pat` will be interpreted as a regex, and thus all special characters for regex will behave differently. Eg `s.str.startswith(".")` will no longer do the correct thing.

Reply:
tests were added. can you show current output and expected.

Reply:
Do we already have CI with pyarrow 4.0 / nightly? 

Didn't try yet through latest pandas, but just checking the example data from the test with pyarrow:

```
In [24]: arr = pa.array(["om", None, "foo_nom", "nom", "bar_foo", None, "foo", "regex", "rege."])

In [25]: pc.match_substring_regex(arr, "^rege.")
Out[25]: 
<pyarrow.lib.BooleanArray object at 0x7fae229d7340>
[
  false,
  null,
  false,
  false,
  false,
  null,
  false,
  true,
  true
]
```

The last two values are both True, while the expected test output has False, True.

Reply:
Whoops, sorry, I completely missed the `re.escape(pat)`. That of course solves my concern ;)